bevy_script_api/src/common/std.rs

```diff
@@ -131,10 +131,11 @@ impl<T: FromReflect + TypePath> IntoIterator for ScriptVec<T> {
             // TODO?: end used to be an Option, and this check moved into the next method but
             // I am not sure if this will ever realistically fail, so if you do get this exception happening
             // hit me with an issue
+            // if len > 0, subtract 1, otherwise set to 0
             end: self
                 .len()
-                .map(|v| v - 1)
```

Comment:
Hmm, I think this will still have the problem that when there are zero elements, the iterator will try to index into the vec when `start == end`.

Let's maybe rename end to len, set it to len here directly to avoid these shenanigans and then change the condition in the iterator to `start < len`

Reply:
I changed the iterator to
```rust
impl<T: FromReflect> Iterator for ScriptVecIterator<T> {
    type Item = ScriptRef;

    fn next(&mut self) -> Option<Self::Item> {
        let nxt = (self.current < self.len).then(|| self.base.index(self.current));
        self.current += 1;
        nxt
    }
}
```
As suggested by you, we now check `start < len`. also `end` has been renamed to `len`.
This no longer throws an error if the vec is empty, but it also doesn't fully loop through the vec:
```text
MyThing.array: [1, 2, 3]
MyThing.array[0]: 1
Looping through MyThing.array:
MyThing.array element: 1
MyThing.array element: 2
```
As you can see, the last element is not looped through.

Reply:
I changed it back to `<=` currently, so that it still loops through all elements

Reply:
oh. yep good point, that somehow flew over my head

Reply:
this seemed to fix it btw!